src/worker.js

```diff
@@ -105,6 +105,14 @@ Module['instantiateWasm'] = (info, receiveInstance) => {
 }
 #endif
 
+// Turn unhandled rejected promises into errors so that the main thread will be
+// notified about them.
+self.onunhandledrejection = (e) => {
```

Comment:
Might be better to use `addEventListener` here so it doesn't interfere with any other global handler the user may have added. We don't do that for `onmessage` etc., which is risky as well, but we already assume we handle all the messaging so maybe that's ok. But users may have their own error handling for promises.

Reply:
I don't think we provide any way for the user to add event listeners in the workers.js do we?    worker.js is kind of designed to "own" then entire works, not run alongside random user code I think?

Reply:
I don't think we provide any way for the user to add event listeners in the workers.js do we?    worker.js is kind of designed to "own" then entire works, not run alongside random user code I think?

Reply:
`--pre-js` code ends up evalled into the worker.js scope, I think? So any hooks there would end up relevant.

Reply:
I guess the user is more likely to want to add a promise rejection handler than an onmessage handler? But if they do, they probably want to override this one, so maybe they can do that in --post-js and it will be ok?

Reply:
Hmm, good point, maybe the user being able to override this one is better actually. sgtm.

Reply:
This is a general change to all pthreads-using applications, though, which is what worries me. Gating this on WasmFS would be safer, but odd. Perhaps we should at least document this in the changelog.

Reply:
Surely this is good change that all programs will want.   Without this, such errors go un-detected and un-reported.  I see no reason to make this wasmfs related given that it relates to all JS promises.

Reply:
Agreed. Changelog mention seems enough.